__test__/pet.test.js

```diff
@@ -0,0 +1,155 @@
+    });
+});
+
+describe('constructor', () => {
+    it('increases the hunger property by 5', () => {
+        const pet = new Pet('Fido');
+        pet.growUp();
+        expect(pet.hunger).toEqual(5);
+    });
+});
+
+describe('constructor', () => {
+    it('decreases fitness by 3', () => {
+        const pet = new Pet('Fido');
+        pet.fitness = 10;
+        pet.growUp();
+        expect(pet.fitness).toEqual(7);
+    });
+});
+
+describe('walk', () => {
+
+    it('throws an error is the pet is not alive', () => {
+        const pet = new Pet('Fido');
+
+        pet.fitness = 0;
+
+        expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+    });
+
+    it('increases fitness by 4 to a maximum of 10', () => {
+        const pet = new Pet('Fido');
+
+        pet.fitness = 8;
+        pet.walk();
+
+        expect(pet.fitness).toEqual(10);
+    });
+});
+
+describe('feed', () => {
+
+    it('throws an error is the pet is not alive', () => {
+        const pet = new Pet('Fido');
+
+        pet.age = 30;
+
+        expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+    });
+
+    it('decreases hunger level by 3 to a minimum of 0', () => {
+        const pet = new Pet('Fido');
+
+        pet.hunger = 5;
+        pet.feed();
+        expect(pet.hunger).toEqual(2);
+    });
+});
+
+describe('checkup', () => {
+    it('tells you if your pet needs walking or not', () => {
+        const pet = new Pet('Fido');
+        pet.fitness = 2;
+        expect(pet.checkup()).toBe("I need a walk");
+    });
+
+    it('tells you if your pet is hungry or not', () => {
+        const pet = new Pet('Fido');
+        pet.hunger = 6;
+        expect(pet.checkup()).toBe("I am hungry");
+    });
+
+    it('tells you if your pet is hungry and needs a walk', () => {
+        const pet = new Pet('Fido');
+        pet.hunger = 6;
+        pet.fitness = 2;
+        expect(pet.checkup()).toBe("I am hungry AND I need a walk");
+    });
+
+    it('tells you if your pet is hungry and needs a walk', () => {
+        const pet = new Pet('Fido');
+        pet.hunger = 3;
+        pet.fitness = 6;
+        expect(pet.checkup()).toBe("I feel great");
+    });
+});
+
+describe('death', () => {
+    it('tells you if your pet is alive - fitness', () => {
+        const pet = new Pet('Fido');
+        pet.fitness = 0;
+        expect(pet.isAlive).toBe(false);
+    });
+
+    it('tells you if your pet is alive - hunger', () => {
+        const pet = new Pet('Fido');
+        pet.hunger = 12;
+        expect(pet.isAlive).toBe(false);
+    });
+
+    it('tells you if your pet is alive - age', () => {
+        const pet = new Pet('Fido');
+        pet.age = 31;
+        expect(pet.isAlive).toBe(false);
+    });
+
+    describe('having a baby', () => {
+        it('has a baby', () => {
+            const parent = new Pet("Fido");
+            parent.haveBaby('Amelia');
+
+            expect(parent.children[0].name).toEqual("Amelia");
+
```

Comment:
I would add another expect statement here that checks the `parent.children[0]` is a instance of Pet i.e.
```
const parent = new Pet("Fido");
parent.haveBaby('Amelia');

expect(parent.children[0].name).toEqual('Amelia');
expect(parent.children[0]).toBeInstanceOf(Pet);
```
This just makes this test a little more robust, as atm you could pass this test by having your child just be an object with a name property rather than an instance of the Pet constructor.